Pull in r200452 from upstream clang trunk (by Jakob Stoklund Olesen):

  SPARCv9 supports atomic operations up to 64 bits.

  Patch by Roman Divacky!

Introduced here: http://svnweb.freebsd.org/changeset/base/262262

Index: tools/clang/lib/Basic/Targets.cpp
===================================================================
--- tools/clang/lib/Basic/Targets.cpp
+++ tools/clang/lib/Basic/Targets.cpp
@@ -4535,6 +4535,7 @@ class SparcV9TargetInfo : public SparcTargetInfo {
     LongDoubleWidth = 128;
     LongDoubleAlign = 128;
     LongDoubleFormat = &llvm::APFloat::IEEEquad;
+    MaxAtomicPromoteWidth = MaxAtomicInlineWidth = 64;
   }
 
   virtual void getTargetDefines(const LangOptions &Opts,
